Compilation error with Criteria

Following this PR: https://github.com/jhipster/generator-jhipster/pull/8054
It seems to broke all current builds, after v5.3.3 release.

I look Travis, and here the log errors:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 38.714 s
[INFO] Finished at: 2018-09-18T03:34:48+10:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project travis-default: Compilation failure: Compilation failure:
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[194,5] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[197,24] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[212,9] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[216,27] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/BankAccountCriteria.java:[234,2] reached end of file while parsing
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[432,5] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[435,24] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[471,9] illegal start of expression
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[475,27] ';' expected
[ERROR] /home/travis/app/src/main/java/io/github/jhipster/travis/service/dto/FieldTestServiceClassEntityCriteria.java:[514,2] reached end of file while parsing
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
mv: cannot stat ‘target/*.war’: No such file or directory
Error when packaging
```

--- a/generators/entity-server/templates/src/main/java/package/service/dto/EntityCriteria.java.ejs
+++ b/generators/entity-server/templates/src/main/java/package/service/dto/EntityCriteria.java.ejs
@@ -119,7 +119,7 @@ public class <%= entityClass %>Criteria implements Serializable {
     public int hashCode() {
         return Objects.hash(
 <%_ filterVariables.forEach((filterVariable, index) => { _%>
-    <%_ if (filterVariables.length == (index - 1)) { _%>
+    <%_ if (filterVariables.length === (index + 1)) { _%>
         <%= filterVariable.name %>
     <%_ } else { _%>
         <%= filterVariable.name %>,

